tests/test_core_api_base.py

```diff
@@ -0,0 +1,297 @@
+                    end=Timestamp("2016-02-05 01:00:00"),
+                ),
+            ],
+            None,
+            None,
+            Timedelta(hours=2),
+            [
+                Event(
+                    begin=Timestamp("2016-02-05 00:00:00"),
+                    end=Timestamp("2016-02-05 02:00:00"),
+                ),
+            ],
+            id="duration_exceeds_files_duration",
+        ),
+        pytest.param(
+            [
+                BaseFile(
+                    path=Path("foo"),
+                    begin=Timestamp("2016-02-05 00:00:00"),
+                    end=Timestamp("2016-02-05 01:00:00"),
+                ),
+            ],
+            Timestamp("2016-02-05 00:30:00"),
+            Timestamp("2016-02-05 00:31:00"),
+            None,
+            [
+                Event(
+                    begin=Timestamp("2016-02-05 00:30:00"),
+                    end=Timestamp("2016-02-05 00:31:00"),
+                ),
+            ],
+            id="one_file_one_data_explicit_begin_end_parts",
+        ),
+        pytest.param(
+            [
+                BaseFile(
+                    path=Path("foo"),
+                    begin=Timestamp("2016-02-05 00:00:00"),
+                    end=Timestamp("2016-02-05 01:00:00"),
+                ),
+            ],
+            Timestamp("2016-02-05 00:30:00"),
+            Timestamp("2016-02-05 00:31:00"),
+            Timedelta(seconds=20),
+            [
+                Event(
+                    begin=Timestamp("2016-02-05 00:30:00"),
+                    end=Timestamp("2016-02-05 00:30:20"),
+                ),
+                Event(
+                    begin=Timestamp("2016-02-05 00:30:20"),
+                    end=Timestamp("2016-02-05 00:30:40"),
+                ),
+                Event(
+                    begin=Timestamp("2016-02-05 00:30:40"),
+                    end=Timestamp("2016-02-05 00:31:00"),
+                ),
+            ],
+            id="one_file_one_data_explicit_begin_end_duration_parts",
+        ),
+        pytest.param(
+            [
+                BaseFile(
+                    path=Path("foo"),
+                    begin=Timestamp("2016-02-05 00:00:00"),
+                    end=Timestamp("2016-02-05 00:50:00"),
+                ),
+            ],
+            None,
+            None,
+            Timedelta(minutes=20),
+            [
+                Event(
+                    begin=Timestamp("2016-02-05 00:00:00"),
+                    end=Timestamp("2016-02-05 00:20:00"),
+                ),
+                Event(
+                    begin=Timestamp("2016-02-05 00:20:00"),
+                    end=Timestamp("2016-02-05 00:40:00"),
+                ),
+                Event(
+                    begin=Timestamp("2016-02-05 00:40:00"),
+                    end=Timestamp("2016-02-05 01:00:00"),
+                ),
+            ],
+            id="one_file_several_data_with_modulo",
+        ),
```

Comment:
For example, this test wouldn't pass before the changes in this PR because the lase data (from `2016-02-05 00:40:00` to `2016-02-05 01:00:00`) would be missing